components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java

```diff
@@ -3270,6 +3277,11 @@ public Response importOpenAPIDefinition(InputStream fileInputStream, Attachment
             throw RestApiUtil.buildBadRequestException("Error while parsing 'additionalProperties'", e);
         }
 
+        // validate sandbox and production endpoints
+        if (!PublisherCommonUtils.validateEndpoints(apiDTOFromProperties)) {
+            RestApiUtil.handleBadRequest("Invalid/Malformed endpoint URL(s) detected ", log);
```

Comment:
Throw the APIManagementException from the method to the upper level
```suggestion
                            throw new APIManagementException("Invalid/Malformed endpoint URL(s) detected",
                        ExceptionCodes.INVALID_ENDPOINT_URL);
```

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/10715/commits/d2955ccba1428a37539279ca49f5dce722f64b1e